### PR TITLE
Fix redirecting already-registered users from iframe dialog

### PR DESCRIPTION
**Pull request by laughinghan**
One thing missing from #126/#143, which intro'd the feature of
redirecting already-signed-up users to Slack, was support for the iframe
dialog that the "Realtime Badge" button thing pops up:
https://github.com/rauchg/slackin#realtime-badge

In simpler times this would've been a 4-character fix:
```diff
-        location.href = res.body.redirectUrl;
+        top.location.href = res.body.redirectUrl;
```
But nowadays people have futile "policies" trying to be "secure" about
their "content" and "sand" "box" "our" "i" "frames":
[image](http://output.jsbin.com/hagibe)
(You can test this by including `<meta http-equiv="Content-Security-Policy" content="sandbox allow-scripts allow-same-origin">` in the `<head>` of a page containing the slackin badge)

The bottom line is, `top.location.href` doesn't pass muster anymore for a library that's going
to run in arbitrary pages on arbitrary servers. Hence the `postMessage()` RPC hack.

### CODE REVIEW

File: lib/assets/client.js [hunk 1]
```diff
@@ -83,3 +83,34 @@ function anim(el, c){
...
+  // A: Ah, the halcyon days, when scripts in iframes could just arbitrarily
+  //    fuck with parent windows' globals. I'm afraid that in this day and
+  //    age of [sandboxes][] and [Content Security Policies][CSP], it's all
+  //    too common for meddling idiots to try to stop us from doing whatever
+  //    we please with their globals---how dare they trample on our rights.
+  //    I mean we're still allowed to do almost anything to their globals,
+  //    the only global we're not allowed to change now is the location URL
+  //    (seriously). I momentarily hoped that explicit permission from the
+  //    parent of the iframe could ensure this privilege that we deserve of
+  //    redirecting the parent, but [nope][].
```
Comment (rauchg) [inline lib/assets/client.js]: Amazing
